src/SprykerEco/Zed/Braintree/Communication/Plugin/Checkout/BraintreePlaceOrderPreCheckPlugin.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\Braintree\Communication\Plugin\Checkout;
+
+use Generated\Shared\Transfer\BraintreeTransactionResponseTransfer;
+use Generated\Shared\Transfer\CheckoutErrorTransfer;
+use Generated\Shared\Transfer\CheckoutResponseTransfer;
+use Generated\Shared\Transfer\QuoteTransfer;
+use Spryker\Zed\CheckoutExtension\Dependency\Plugin\CheckoutPreConditionPluginInterface;
+use Spryker\Zed\Kernel\Communication\AbstractPlugin;
+use SprykerEco\Shared\Braintree\BraintreeConfig;
+
+/**
+ * @method \SprykerEco\Zed\Braintree\Business\BraintreeFacadeInterface getFacade()
+ * @method \SprykerEco\Zed\Braintree\BraintreeConfig getConfig()
+ * @method \SprykerEco\Zed\Braintree\Persistence\BraintreeQueryContainerInterface getQueryContainer()
+ * @method \SprykerEco\Zed\Braintree\Communication\BraintreeCommunicationFactory getFactory()
+ */
+class BraintreePlaceOrderPreCheckPlugin extends AbstractPlugin implements CheckoutPreConditionPluginInterface
+{
+    /**
+     * {@inheritDoc}
+     * Specification:
+     * - Checks a condition before the order is placed. If the condition fails, an error is added to the response transfer and 'false' is returned.
+     * - Check could be passed (returns 'true') along with errors added to the checkout response.
+     * - Don't use this plugin to write to a DB.
+     *
+     * @api
+     *
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     * @param \Generated\Shared\Transfer\CheckoutResponseTransfer $checkoutResponseTransfer
+     *
+     * @return bool
+     */
+    public function checkCondition(
+        QuoteTransfer $quoteTransfer,
+        CheckoutResponseTransfer $checkoutResponseTransfer
+    ): bool {
+        $quoteTransfer->requirePayment();
```

Comment:
The plugin method contains a business logic. That's not allowed according to our standards. We need to move all this logic behind facade.

Reply:
done